eq_cir_converter_service/services/schema/schema_processor_service.py

```diff
@@ -1,12 +1,44 @@
 """This module converts the schema from the current to the target version."""
 
+from collections.abc import Mapping, Sequence
+from typing import Any
+
+from jsonpath_ng.ext import parse
+
 from eq_cir_converter_service.config.logging_config import logging
+from eq_cir_converter_service.services.schema.extractable_strings import (
+    EXTRACTABLE_STRINGS,
+)
 from eq_cir_converter_service.types.custom_types import ConvertedSchema, InputSchema
+from eq_cir_converter_service.utils.helper_utils import process_element
 
 logger = logging.getLogger(__name__)
 
 
-async def convert_schema(current_version: str, target_version: str, schema: InputSchema) -> ConvertedSchema:
+# --- JSONPath-Based Transformation ---
+def transform_json_schema(schema: Mapping[str, Any], paths: Sequence[Mapping[str, str]]) -> Mapping[str, Any]:
+    """Transforms the JSON schema based on the provided JSONPath expressions."""
+    for path_expr in paths:
+        if isinstance(path_expr, dict) and "json_path" in path_expr:
```

Comment:
What I can see in these json_paths is a repeating pattern: at the very end it is either a "list"(as a schema "element" not python data type), "title", "description" or some minor types like question, label etc. 
Can you divide these based on that? so if "list", if "title", if "description", and then else?
This would make it more human readable (if used instead of "isinstance" method) in future more maintainable and easier for the reviewer?
Can we eliminate some more extractable strings? Possibly if we ask the team responsible for CIR or someone senior? This would make the logic a lot easier the more scenarios we eliminate.

Reply:
Also, `extractable_strings` name is not the ideal name for it as we used to extract them in eq-translations but we don't now.